src/mlpack/methods/radical/radical.cpp

```diff
@@ -188,6 +188,6 @@ void mlpack::radical::WhitenFeatureMajorMatrix(const mat& matX,
   mat matU, matV;
   vec s;
   arma::svd(matU, s, matV, cov(matX));
-  matWhitening = matU * diagmat(1 / sqrt(s)) * trans(matV);
+  matWhitening = matU * diagmat(s) * trans(matV);
```

Comment:
So if we multiply by the singular values, we are just reconstructing the original matrix instead of whitening it.  If what you are seeing is that the singular values are extremely small, then what we can do is try something like `diagmat(1 / (sqrt(s) + 1e-20))` or some other thing like this to prevent division by 0.